docs/LocationInfo.md

```diff
@@ -0,0 +1,85 @@
+3. [MLIR File](#MLIR-file)
+
+## Summary
+Support of Location info propagation in transformation is one of the attractive features of MLIR. onnx-mlir can takes advantage of this feature in compiler transformation, and runtime debugging. This document describes how to maintain and use the location info in onnx-mlir.  program of the In onnx-mlir, there are three types of tests to ensure correctness of implementation:
+- All onnx-mlir transformations are required to propagate the location info from the source to the target
+- Create localtion info when an ONNX model is imported. If there is `onnx_node_name` string attribute for an opteration, the string is transferred to its location. Otherwise, Unknown location is used.
+- MLIR adds file location (in form of filename:line:column) to nodes when reading in a MLIR file, unless the MLIR file already contains location.
+- Use the flag `--preserveLocations` to turn on location info in the output.
+- With the previous two combined, we can track the source of error by dumping out the MLIR file(without `--preserveLocations`) at desired stage (for example, EmitONNXIR, or EmitMLIR), and then continuing transformation by loading the dumped file. The location info will be line number for that dumped file, providing more details than just from the onnx model. 
+
+## ONNX model
+When reading an ONNX model (.onnx file), onnx-mlir tries to attach location info to the generated IR. 
+Some ONNX exporter annotates every operation with an StringAttr, "onnx_node_name", with an unique string for that operation. 
+The importer of onnx-mlir converts the "onnx_node_name" attribute in the ONNX file tostring location info for the operation.
+If the ONNX model does not have "onnx_node_name" attribute, Unknown location is attached.
+
+### Example with onnx_node_name
+
+This roberta  model is downloaded from onnx model zoo. Compile it with command
+`onnx-mlir roberta-base-11.onnx --preserveLocations --EmitONNXBasic`.
+The location info will be displayed in the output when the flag `--preserveLocations` is used.
+In the output file(roberta-base-11.onnx.mlir), two nodes and their location info are shown below.
+
+```
+    %392 = "onnx.Sub"(%390, %391) {onnx_node_name = "Sub_109"} : (tensor<*xf32>, tensor<*xf32>) -> tensor<*xf32> loc(#loc393)
+    %393 = onnx.Constant dense<2.000000e+00> : tensor<f32> loc(#loc394)
+
+#loc393 = loc("Sub_109")
+#loc394 = loc("Constant_110")
+```
+The 'onnx_node_name` attribute is only for operations, not for the constant. The location info for Constant is given by the importer.
+
+### Example without onnx_node_name
+The following model, test_add.onnx, came from onnx backend test. It is not from
+onnx exporter and does not have `onnx_node_name` attribute.
+The output of command `onnx-mlir test_add.onnx --preserveLocations --EmitONNXBasic`:
+
+```
+#loc = loc(unknown)
+
+...
+    %0 = "onnx.Add"(%arg0, %arg1) : (tensor<3x4x5xf32>, tensor<3x4x5xf32>) -> tensor<3x4x5xf32> loc(#loc)
```

Comment:
With [a recent change](https://github.com/onnx/onnx-mlir/pull/2518), `--EmitONNXIR` will always produce ops with `onnx_node_name`. Not sure if it is useful to use `--EmitONNXIR` instead of `--EmitONNXBasic` in this example.

Reply:
I think that we should use Location instead of `onnx_node_name` inside onnx-mlir, since Location is maintained by MLIR. We do not need to add a new thing for the similar purpose.